lib/open_project/scm/manageable_repository.rb

```diff
@@ -30,32 +30,47 @@
 module OpenProject
   module Scm
     module ManageableRepository
-      ##
-      # We let SCM vendor implementation define their own
-      # types (e.g., for differences in the management of
-      # local vs. remote repositories).
-      #
-      # But if they are manageable by OpenProject, they must
-      # expose this type through +available_types+.
-      MANAGED_TYPE = :managed
-      ##
-      # Reads from configuration whether new repositories of this kind
-      # may be managed from OpenProject.
-      def manageable?
-        !managed_root.nil?
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      module ClassMethods
+        ##
+        # We let SCM vendor implementation define their own
+        # types (e.g., for differences in the management of
+        # local vs. remote repositories).
+        #
+        # But if they are manageable by OpenProject, they must
+        # expose this type through +available_types+.
+        def managed_type
+          :managed
+        end
+
+        ##
+        # Reads from configuration whether new repositories of this kind
+        # may be managed from OpenProject.
+        def manageable?
+          ! (disabled_types.include?(managed_type) || managed_root.nil?)
```

Comment:
Probably I just don't understand the callers/context of this method. My reasoning foots on the following assumptions:

1. `manageable?` tells me whether the SCM can be managed in general, i.e. if it has the *capability of being managed
2. `available_types` tells me, which capabilities of a repo should be exposed to the user
3. Code facing the user and providing options to him will only care about `available_types`
4. Other code (e.g. the one checking whether managed repos are *supported* at all) uses the `manageable?` check

This has some implications, though. E.g. I could imagine that enabling `:managed` first, then disabling it (after users started using it) would behave differently in the two cases. In my proposal it would probably work on, because the code responsible for "keeping things running" won't care for `available_types`. In your version managed repositories might become unavailable because the SCM adapter is not `manageable?` anymore.

However, these are all assumptions (e.g. that any code cares about `manageable?`).

I hope my input is enough for you to judge, what is apropriate here ^^